### DISCUSSION TITLE
Pytest test working with different versions of program

### DISCUSSION THREAD

**Discussion by hmfkzh** [pytest-dev/pytest#10175] [Q&A] [answered]
I have multiple tests which would work on different versions of my program.

For example testA is working only for versions 2,3,4 and not 5 and later. the other test is working for the test from version 4 and later. based on the pytest documentation, I can create a marker similar to below:

    # content of test_mymodule.py
    import mymodule
    
    minversion3 = pytest.mark.skipif(
        myprogram.__versioninfo__ < 3, reason="at least version 3 is required"
    )
    
    
    @minversion3
    def test_function():
        ...

minversion3 marks the tests to be run provided that program has at least version 3. 
I would like to parametrize it so that I can have something like this:

    @minmaxversion(3.2, 6.1)
    def test_function():
        ...

so that this test is only working for programs with version minimum 3.2 and maximum 6.1.

**Comment by nicoddemus** [comment]
Hi,

That's possible, but you will need to handle your marker yourself... easiest done by using an autouse fixture which inspects the mark and skips accordingly:

```python
@pytest.fixture(autouse=True)
def check_app_version_mark(request):
    if m := request.get_closest_marker("minmaxversion"):
        min_ver, max_ver = m.args
        if not (min_ver <= app.version <= max_ver):
            pytest.skip(reason=f"Outside valid version range: {app.version}")
```

Then you use it like:

```python
@pytest.mark.minmaxversion(3.2, 6.1)
def test_function():
    ...
```

**Comment by The-Compiler** [comment, marked as answer]
A perhaps slightly simpler way to what @nicoddemus has shown would be to write a function returning the appropriate `skipif` mark, instead of hardcoding one to a variable:

```python
def minmaxversion(min_str, max_str):
    minver = tuple(map(int, min_str.split(".")))
    maxver = tuple(map(int, max_str.split(".")))
    return pytest.mark.skipif(
        not minver <= myprogram.__versioninfo__ <= maxver,
        reason=f"versions {minver} to {maxver} are required"
    )

@minmaxversion("3.1", "6.1")
def test_function():
    ...
```

(your `3.2` and `6.1` would be floats, which are a weird way to represent version numbers - also, this is completely untested)
